### PR TITLE
Update Simplified Chinese

### PR DESCRIPTION
**Pull request by sbzlzh**

### CODE REVIEW

**Review by TimGoll — APPROVED** [review]

**Comment by sbzlzh** [comment]
@TheOnly8Z Take a look when you have time.

**Comment by sbzlzh** [comment]
@TimGoll Let's see if this works, I solved a bug that appeared inadvertently during F1, is there another way .

**Comment by TimGoll** [comment]
@ZenBre4ker could you take a look at this Bugfix. Is this something you were working on?

@sbzlzh for the future: these two things should be in two separate PRs. It is fine for this one though

**Comment by TimGoll** [comment]
> > @ZenBre4ker could you take a look at this Bugfix. Is this something you were working on?
> > @sbzlzh for the future: these two things should be in two separate PRs. It is fine for this one though
> 
> I also think we should split it up, but I'm a little overwhelmed, so what about another branch?

Keep it for now. But for the future yes, two branches would work. This is how we do it

**Comment by sbzlzh** [comment]
> > > @ZenBre4ker could you take a look at this Bugfix. Is this something you were working on?
> > > @sbzlzh for the future: these two things should be in two separate PRs. It is fine for this one though
> > 
> > 
> > I also think we should split it up, but I'm a little overwhelmed, so what about another branch?
> 
> Keep it for now. But for the future yes, two branches would work. This is how we do it

very nice

**Comment by ZenBre4ker** [comment]
Looks like a good catch and makes sense. Dont think my other PRs change any of it. ✌️

**Comment by TimGoll** [comment]
> Looks like a good catch and makes sense. Dont think my other PRs change any of it. ✌️

Perfect

**Comment by TimGoll** [comment]
@sbzlzh please add a note in the cgangelog file about what you fixed

**Comment by sbzlzh** [comment]
> @sbzlzh please add a note in the cgangelog file about what you fixed

ok

File: lua/terrortown/menus/gamemode/administration/playermodels.lua [hunk 1]
```diff
@@ -63,43 +63,46 @@ function CLGAMEMODESUBMENU:Populate(parent)
 			end
 		}, base)
 
-		playermodels.IsSelectedModel(name, function(value)
-			boxCache[name]:SetModelSelected(value, false)
-		end)
+		-- Add additional checks to ensure boxCache[name] exists
+		if boxCache[name] then
```
Comment (ZenBre4ker) [inline lua/terrortown/menus/gamemode/administration/playermodels.lua:67]: This check should be unnecessary. We created boxcache{name] a few lines before.  You rather need to check it in the callback functions, but I believe that they are synced anyways.

**Review by ZenBre4ker — CHANGES_REQUESTED** [review]
Same as Tim and move that one check. Rest seems fine.

File: lua/terrortown/menus/gamemode/administration/playermodels.lua (hunk 1 above)
Comment (ZenBre4ker) [reply lua/terrortown/menus/gamemode/administration/playermodels.lua:67]: Just checked it, playermodels are not necessarily synced. So move the boxCache[name] check into playermodels.IsSelectedModel and playermodels.IsHattableModel.

**Comment by ZenBre4ker** [comment]
And if you already change slider and checkbox you can update dcombobox_ttt2 and dtextentry as well.

File: lua/terrortown/menus/gamemode/administration/playermodels.lua (hunk 1 above)
Comment (sbzlzh) [reply lua/terrortown/menus/gamemode/administration/playermodels.lua:67]: Is that so?

**Comment by sbzlzh** [comment]
> And if you already change slider and checkbox you can update dcombobox_ttt2 and dtextentry as well.

How to update it : (

**Comment by sbzlzh** [comment]
@ZenBre4ker Is there anything else you need to do?

File: gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dnumslider_ttt2.lua [hunk 2]
```diff
@@ -313,7 +313,8 @@ function PANEL:SetServerConVar(cvar)
 	self.serverConVar = cvar
 
 	cvars.ServerConVarGetValue(cvar, function (wasSuccess, value, default)
-		if wasSuccess then
+		-- Check if self is valid before calling SetValue.
+		if wasSuccess and IsValid(self) then
```
Comment (ZenBre4ker) [inline gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dnumslider_ttt2.lua:317]: Same check could be done in dcombobox_ttt2.lua and dtextentry for cvars.ServerConVarGetValue.

File: lua/terrortown/menus/gamemode/administration/playermodels.lua (hunk 1 above)
Comment (ZenBre4ker) [reply lua/terrortown/menus/gamemode/administration/playermodels.lua:67]: Please dont resolve comments yourself.
And this is not what I meant. you dont need to check boxCache[name] before adding the anonymous function, you need to check it inside of the anonymous function.

File: lua/terrortown/menus/gamemode/administration/playermodels.lua [hunk 3]
```diff
@@ -63,43 +63,46 @@ function CLGAMEMODESUBMENU:Populate(parent)
...
-		playermodels.IsHattableModel(name, function(value)
-			boxCache[name]:SetModelHattable(value, false)
-		end)
+			playermodels.IsHattableModel(name, function(value)
+				boxCache[name]:SetModelHattable(value, false)
+			end)
+		end
 
-		playermodels.RemoveChangeCallback(name, playermodels.state.selected, "TTT2F1MenuPlayermodels")
+			playermodels.RemoveChangeCallback(name, playermodels.state.selected, "TTT2F1MenuPlayermodels")
```
Comment (ZenBre4ker) [inline lua/terrortown/menus/gamemode/administration/playermodels.lua:77]: Correct indentation again.

**Review by ZenBre4ker — CHANGES_REQUESTED** [review]
Well you didnt change what I mean. I suggest you remove the changes to the other files and only update the chinese with this PR here.
The other changes make more sense under a separate PR.

File: gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dnumslider_ttt2.lua (hunk 2 above)
Comment (sbzlzh) [reply gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dnumslider_ttt2.lua:317]: all right

File: lua/terrortown/menus/gamemode/administration/playermodels.lua (hunk 3 above)
Comment (sbzlzh) [reply lua/terrortown/menus/gamemode/administration/playermodels.lua]: ok

**Review by ZenBre4ker — APPROVED** [review]
Thanks a lot.

**Comment by TEGTainFan** [comment]
permit

**Pull request merged**